src/common.c

```diff
@@ -437,7 +437,7 @@ static void set_flash_cr_pg(stlink_t *sl) {
         x |= 1 << FLASH_CR_PG;
     } else if (sl->flash_type == STLINK_FLASH_TYPE_L4) {
         cr_reg = STM32L4_FLASH_CR;
-        x &= ~STM32L4_FLASH_CR_OPBITS;
+        x &= (uint32_t)~STM32L4_FLASH_CR_OPBITS;
         x |= 1 << STM32L4_FLASH_CR_PG;
```

Comment:
Maybe it is more useful to make definition of STM32L4_FLASH_CR_OPBITS typed?
For example like this : 
```cpp
#define STM32L4_FLASH_CR_OPTLOCK  30u	/* Lock option bytes */
```

or

```cpp
#define STM32L4_FLASH_CR_OPTLOCK  (uint32_t)30	/* Lock option bytes */
```

Reply:
Indeed better, changed